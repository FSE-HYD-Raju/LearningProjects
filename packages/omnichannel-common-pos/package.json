{
	"name": "omnichannel-common-pos",
	"version": "111.2.0",
	"description": "Common actions, stores and components for omnichannel",
	"main": "lib/index",
	"types": "lib/index.d.ts",
	"oc:es6": "src/index",
	"scripts": {
		"test": "../../testSinglePackage.sh $(pwd)",
		"test-watch": "jest --watch",
		"lint": "eslint tests src --max-warnings=0 ",
		"test-ci": "jest",
		"lint-fix": "eslint src tests --fix",
		"flow": "flow; test $? -eq 0 -o $? -eq 2",
		"clean": "rm -rf lib",
		"build-ts": "tsc --pretty",
		"copy-assets": "copyfiles --up 1 'src/**/*.@(scss|jpg|jpeg|gif|png|svg)' ./lib",
		"build": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -s inline -d lib/ src/ && npm run build-ts && npm run copy-assets && rimraf -rf messages",
		"watch": "cross-env NODE_ENV=production BABEL_ENV=production BABEL_DISABLE_CACHE=1 babel -w -s inline -d lib/ src/ && rimraf -rf messages",
		"prepublish": "npm run clean && npm run build"
	},
	"jest": {
		"testRegex": "(/tests/.*)\\.test\\.js$",
		"coverageDirectory": "jestcoverage",
		"coverageReporters": [
			"lcov",
			"html"
		],
		"setupFiles": [
			"../../jest-setup.js"
		]
	},
	"author": "",
	"license": "ISC",
	"publishConfig": {
		"registry": "https://artifactory.qvantel.net/artifactory/api/npm/npm-local"
	},
	"dependencies": {
		"alt": "0.18.6",
		"alt-utils": "2.0.0",
		"autosuggest-highlight": "2.1.1",
		"axios": "0.18.0",
		"babel-cli": "6.26.0",
		"babel-core": "6.26.0",
		"babel-eslint": "7.2.3",
		"babel-jest": "23.6.0",
		"babel-loader": "7.1.3",
		"babel-plugin-add-module-exports": "0.2.1",
		"babel-plugin-module-resolver": "3.1.0",
		"babel-plugin-react-intl": "2.4.0",
		"babel-plugin-react-transform": "2.0.2",
		"babel-plugin-transform-class-properties": "6.24.1",
		"babel-plugin-transform-decorators-legacy": "1.3.4",
		"babel-plugin-transform-es2015-classes": "6.24.1",
		"babel-plugin-transform-proto-to-assign": "6.26.0",
		"babel-plugin-transform-react-constant-elements": "6.23.0",
		"babel-plugin-transform-react-inline-elements": "6.22.0",
		"babel-plugin-transform-react-pure-class-to-function": "1.0.1",
		"babel-plugin-transform-react-remove-prop-types": "0.4.20",
		"babel-plugin-transform-runtime": "6.23.0",
		"babel-plugin-typecheck": "3.9.0",
		"babel-polyfill": "6.26.0",
		"babel-preset-es2015": "6.24.1",
		"babel-preset-es2015-loose": "7.0.0",
		"babel-preset-react": "6.24.1",
		"babel-preset-stage-0": "6.24.1",
		"babel-register": "6.26.0",
		"babel-runtime": "6.26.0",
		"bootstrap-sass": "3.3.7",
		"byte-size": "4.0.4",
		"classnames": "2.2.6",
		"connected-react-router": "4.3.0",
		"copyfiles": "2.0.0",
		"country-data": "0.0.31",
		"cross-env": "5.1.4",
		"crypto-js": "3.1.9-1",
		"css-ns": "1.2.1",
		"debug": "4.1.0",
		"diff": "3.4.0",
		"history": "4.7.2",
		"jwt-decode": "2.2.0",
		"langs": "2.0.0",
		"lodash": "4.17.11",
		"moment": "2.21.0",
		"moment-duration-format": "2.2.2",
		"object-assign": "4.1.1",
		"omnichannel-flow-pos": "111.2.0",
		"path": "0.12.7",
		"path-to-regexp": "2.3.0",
		"prop-types": "15.6.1",
		"ramda": "0.24.1",
		"rc-progress": "2.2.5",
		"rc-slider": "8.6.1",
		"react": "16.10.2",
		"react-autosuggest": "9.3.4",
		"react-bootstrap": "0.32.1",
		"react-collapse": "4.0.3",
		"react-confirm": "0.1.14",
		"react-dates": "20.1.0",
		"react-dom": "16.10.2",
		"react-dropzone": "3.13.4",
		"react-flatpickr": "3.6.4",
		"react-formal": "0.25.4",
		"react-formal-inputs": "2.1.1",
		"react-helmet": "5.2.0",
		"react-highlight-words": "0.11.0",
		"react-input-range": "1.3.0",
		"react-intl": "2.7.0",
		"react-motion": "0.5.2",
		"react-motion-ui-pack": "0.10.3",
		"react-onclickoutside": "6.7.1",
		"react-redux": "5.0.7",
		"react-router": "4.2.0",
		"react-router-dom": "4.2.2",
		"react-select": "1.2.1",
		"react-toastr": "3.0.0",
		"react-widgets": "3.5.0",
		"recompose": "0.30.0",
		"redux": "4.0.0",
		"redux-saga": "0.16.0",
		"redux-saga-tester": "1.0.379",
		"shallowequal": "0.2.2",
		"twix": "1.2.1",
		"universal-cookie": "2.1.2",
		"uuid": "3.2.1",
		"weasel-styles": "2.3.1",
		"yup": "0.21.3"
	},
	"devDependencies": {
		"@types/alt": "0.16.35",
		"@types/classnames": "2.2.6",
		"@types/jest": "23.3.9",
		"@types/lodash": "4.14.118",
		"@types/moment-duration-format": "2.2.2",
		"@types/node": "8.0.0",
		"@types/rc-slider": "8.2.3",
		"@types/react": "16.7.18",
		"@types/react-bootstrap": "0.32.14",
		"@types/react-collapse": "4.0.2",
		"@types/react-confirm": "0.1.2",
		"@types/react-dates": "17.1.5",
		"@types/react-dom": "16.0.5",
		"@types/react-flatpickr": "3.7.0",
		"@types/react-helmet": "5.0.7",
		"@types/react-highlight-words": "0.12.0",
		"@types/react-intl": "2.3.9",
		"@types/react-motion": "0.0.27",
		"@types/react-motion-ui-pack": "0.10.1",
		"@types/react-onclickoutside": "6.0.4",
		"@types/react-redux": "6.0.0",
		"@types/react-router": "4.0.25",
		"@types/react-select": "1.2.0",
		"@types/react-toastr": "3.0.3",
		"@types/recompose": "0.26.4",
		"@types/redux-actions": "2.3.0",
		"@types/redux-devtools": "3.0.44",
		"@types/redux-devtools-dock-monitor": "1.1.32",
		"@types/redux-devtools-log-monitor": "1.0.33",
		"@types/redux-logger": "3.0.6",
		"@types/seamless-immutable": "7.1.2",
		"@types/universal-cookie": "2.2.0",
		"@types/webpack": "4.4.0",
		"@types/webpack-env": "1.13.6",
		"@types/yup": "0.26.0",
		"babel-plugin-__coverage__": "11.0.0",
		"babel-plugin-transform-flow-strip-types": "6.22.0",
		"babel-plugin-transform-runtime": "6.23.0",
		"enzyme": "3.7.0",
		"enzyme-adapter-react-16": "1.10.0",
		"enzyme-to-json": "3.3.3",
		"eslint": "4.10.0",
		"eslint-config-omnichannel": "2.2.4",
		"eslint-config-prettier": "1.7.0",
		"eslint-config-react-app": "2.1.0",
		"eslint-loader": "1.9.0",
		"eslint-plugin-babel": "4.1.2",
		"eslint-plugin-flowtype": "2.39.1",
		"eslint-plugin-import": "2.8.0",
		"eslint-plugin-jsx-a11y": "5.1.1",
		"eslint-plugin-prettier": "2.6.0",
		"eslint-plugin-react": "7.4.0",
		"eslint-plugin-react-intl": "1.1.1",
		"flow-bin": "0.70.0",
		"jest": "23.6.0",
		"jest-junit": "3.6.0",
		"lint-staged": "4.3.0",
		"moxios": "0.4.0",
		"pre-commit": "1.2.2",
		"prettier": "1.6.1",
		"react-router-enzyme-context": "1.2.0",
		"react-test-renderer": "16.7.0",
		"redux-devtools": "3.4.1",
		"redux-devtools-dock-monitor": "1.1.3",
		"redux-devtools-log-monitor": "1.4.0",
		"redux-logger": "3.0.6",
		"rimraf": "2.6.2",
		"sinon": "4.3.0",
		"ts-jest": "23.10.5",
		"ts-loader": "5.3.1",
		"typescript": "3.1.6",
		"webpack": "4.26.1"
	},
	"commitInfo": "omni-ui-common-pos|184f176|1571059113|Venkata_Gangadhar_Kodela|develop"
}
