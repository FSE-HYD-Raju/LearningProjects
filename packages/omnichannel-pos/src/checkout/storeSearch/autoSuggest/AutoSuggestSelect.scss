@import '~omnichannel-common-pos/lib/styles/mixins.scss';
@import '~omnichannel-common-pos/lib/styles/_variables.scss';

$height: 50px;
$dropdown-arrow-padding: 50px;

.AutoSuggestSelect {

	position: relative;
	height: $height;
	@extend .OcInput-text;

	.OcInput {
		position: absolute;
		top: 0;
		bottom: 0;
		left: 0;
		right: $dropdown-arrow-padding;
		background-color: transparent;
		border: none;

		.OcInput-text {
			border: none;
			box-shadow: none;
		}

		.OcInput-input:before, .OcInput-input:after {
			display: none;
		}
	}

	&-select {
		height: $height;
		position: absolute;
		top: 0;
		bottom: 0;
		left: 0;
		right: 0;

		.Select-placeholder {
			display: none;
		}

		.Select-control {
			height: 100%;
		}

		.Select-input {
			height: 100%;

			>input {
				height: 100%;
				display: none;
			}
		}

		.Select-option mark {
			color: theme-color("danger");
			font-weight: bold;
			background-color: transparent;
			padding: 0;
		}
		.Select-option.is-focused {
			background-color: theme-color("primary");
			color: $white;
		}
	}
}
