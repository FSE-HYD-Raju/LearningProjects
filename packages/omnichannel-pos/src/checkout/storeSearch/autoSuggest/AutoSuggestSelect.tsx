import { Component } from "react";
import Highlighter from "react-highlight-words";
import { OcInput, cssns } from "omnichannel-common-pos";
import { Async } from "react-select";

const { React } = cssns("AutoSuggestSelect");

interface AutoSuggestSelectProps {
	onSuggestionSelected: (selectedItem: any) => void;
	loadOptions?: (inputValue?: string) => Promise<any>;
	labelField: string;
	idField: string;
	placeholder?: string;
	value?: string;
	required?: boolean;
	minimalLength?: number;
}

const DEFAULT_MIN_LENGTH: number = 3;

interface AutoSuggestSelectState {
	value: string;
}

class AutoSuggestSelect extends Component<AutoSuggestSelectProps, AutoSuggestSelectState> {
	selectComponent: any;
	inputComponent: any;

	constructor(props: AutoSuggestSelectProps) {
		super(props);
		this.state = {
			value: props.value || ""
		};
	}

	componentWillReceiveProps(newProps: AutoSuggestSelectProps) {
		if (this.props.value !== newProps.value) {
			const newState = {
				value: newProps.value || ""
			};
			this.setState(newState);
		}
	}

	onChange = (selectedItem?: any) => {
		this.setState(
			{
				value:
					(selectedItem && selectedItem[this.props.labelField]) || ""
			},
			() => {
				if (this.props.onSuggestionSelected) {
					this.props.onSuggestionSelected(selectedItem);
				}
			}
		);
	};

	onInputChange = (changeEvent?: any) => {
		if (!changeEvent) {
			// clear icon click
			this.updateState();
			if (this.inputComponent) {
				this.inputComponent.inputElement.value = "";
			}
		} else if (changeEvent && changeEvent.target) {
			this.updateState(changeEvent.target.value);
		}
	};

	updateState = (inputText: string = "") => {
		this.setState({ value: inputText }, () => {
			if (this.selectComponent) {
				this.selectComponent.select.handleInputChange({
					target: { value: inputText }
				});
			}
		});
	};

	loadData = (inputText: string): Promise<any> => {
		const minLength = this.props.minimalLength || DEFAULT_MIN_LENGTH;
		if (inputText && inputText !== "" && inputText.length >= minLength) {
			if (this.props.loadOptions) {
				return this.props.loadOptions(inputText);
			}
		}
		return Promise.resolve({ options: [] });
	};

	onMenuClose = () => {
		if (this.inputComponent) {
			this.inputComponent.inputElement.focus();
		}
	};

	renderOption = (item: any) => {
		return (
			<Highlighter
				searchWords={[this.state.value]}
				textToHighlight={item[this.props.labelField]}
			/>
		);
	};

	render() {
		const { labelField, idField, required, placeholder } = this.props;
		const { value } = this.state;

		return (
			<div className="AutoSuggestSelect">
				<OcInput
					ref={inputComponent => {
						this.inputComponent = inputComponent;
					}}
					value={this.state.value}
					onChange={this.onInputChange}
					placeholder={this.props.placeholder}
					required={this.props.required}
				/>
				<Async
					className="AutoSuggestSelect-select"
					ref={(selectComponent: any) => {
						this.selectComponent = selectComponent;
					}}
					autoBlur={true}
					autoload={false}
					backspaceRemoves={false}
					labelKey={labelField}
					openOnFocus={true}
					loadOptions={this.loadData}
					loadingPlaceholder="Loading..."
					multi={false}
					onChange={this.onChange}
					onInputChange={this.onInputChange}
					onBlurResetsInput={false}
					onCloseResetsInput={false}
					onClose={this.onMenuClose}
					placeholder={placeholder}
					searchPromptText=""
					required={required}
					optionRenderer={this.renderOption}
					value={value}
					valueKey={idField}
				/>
			</div>
		);
	}
}

export default AutoSuggestSelect;
