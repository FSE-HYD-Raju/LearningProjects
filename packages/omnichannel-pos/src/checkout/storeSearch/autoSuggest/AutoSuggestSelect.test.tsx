import * as React from "react";
import { mountWithContext, shallowWithContext } from "omnichannel-common-pos";
import AutoSuggestSelect from "./AutoSuggestSelect";

describe("AutoSuggestSelect", () => {
	const testData = {
		options: [
			{ value: "1", label: "Label 1" },
			{ value: "2", label: "Label 2" },
			{ value: "3", label: "Label 3" },
			{ value: "4", label: "Label 4" }
		]
	};

	const minProps = {
		labelField: "label",
		idField: "value",
		onSuggestionSelected: jest.fn(),
		loadOptions: jest.fn(),
	};

	it("succeeds at shallow mount without props", () => {
		const wrapper = shallowWithContext(<AutoSuggestSelect {...minProps} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("should mount without props", () => {
		mountWithContext(<AutoSuggestSelect {...minProps} />);
	});

	it("placeholder is shown when input is empty", () => {
		const props = { ...minProps, placeholder: "Test" };

		const wrapper = mountWithContext(<AutoSuggestSelect {...props} />);

		const placelholderLabel = wrapper.find("OcInput label");
		expect(placelholderLabel.text()).toMatch("Test");
	});

	it("selected item displayed if provided", () => {
		const props = { ...minProps, value: "Default selected value" };

		const wrapper = mountWithContext(<AutoSuggestSelect {...props} />);

		const input = wrapper.find("OcInput input");
		expect(input.instance().value).toMatch("Default selected value");
	});

	it("loadData called after specified count of symbols in input", () => {
		const props = { ...minProps, minimalLength: 2 };
		const loadOptions = minProps.loadOptions;
		const wrapper = mountWithContext(<AutoSuggestSelect {...props} />);
		wrapper
			.find("OcInput input")
			.simulate("change", { target: { value: "te" } });
		expect(loadOptions).toHaveBeenCalled();
	});

	// TODO: this is quite strange but wrapper.update() called on AutoSuggestSelect leads to rendering empty string instead
	// TODO: of real component, this should be investigated additionally
	xit("loaded data stored in selectComponent", async () => {
		const loadDataMock = jest.fn((): Promise<any> => {
			return Promise.resolve(testData);
		});
		const props = { ...minProps, loadOptions: loadDataMock };
		const wrapper = mountWithContext(<AutoSuggestSelect {...props} />);

		await expect(loadDataMock).toHaveBeenCalled();

		expect(wrapper.instance().selectComponent.state.isLoading).toBeFalsy();
		expect(wrapper.instance().selectComponent.state.options).not.toBeNull();
		expect(wrapper.instance().selectComponent.state.options.length).toBe(4);
	});
});
