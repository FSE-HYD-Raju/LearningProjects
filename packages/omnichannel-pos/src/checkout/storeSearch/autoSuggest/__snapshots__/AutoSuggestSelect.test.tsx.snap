// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoSuggestSelect succeeds at shallow mount without props 1`] = `
<div
  className="AutoSuggestSelect"
>
  <OcInput
    onChange={[Function]}
    value=""
  />
  <Async
    autoBlur={true}
    autoload={false}
    backspaceRemoves={false}
    cache={Object {}}
    className="AutoSuggestSelect-select"
    ignoreAccents={true}
    ignoreCase={true}
    labelKey="label"
    loadOptions={[Function]}
    loadingPlaceholder="Loading..."
    multi={false}
    onBlurResetsInput={false}
    onChange={[Function]}
    onClose={[Function]}
    onCloseResetsInput={false}
    onInputChange={[Function]}
    openOnFocus={true}
    optionRenderer={[Function]}
    options={Array []}
    searchPromptText=""
    value=""
    valueKey="value"
  >
    <Component />
  </Async>
</div>
`;
